src/Microsoft.AspNet.Mvc.Core/ActionResults/ViewResult.cs

```diff
@@ -50,10 +51,22 @@ public class ViewResult : ActionResult
             var viewEngine = ViewEngine ??
                              context.HttpContext.RequestServices.GetRequiredService<ICompositeViewEngine>();
 
+            var logger = context.HttpContext.RequestServices.GetRequiredService<ILoggerFactory>()
+                .CreateLogger<ViewResult>();
+
             var viewName = ViewName ?? context.ActionDescriptor.Name;
-            var view = viewEngine.FindView(context, viewName)
-                                 .EnsureSuccessful()
-                                 .View;
+            var viewEngineResult = viewEngine.FindView(context, viewName);
+            if(!viewEngineResult.Success)
+            {
+                logger.LogError(
+                    "The view '{ViewName}' was not found. Searched locations: {SearchedViewLocations}", 
+                    viewName,
+                    viewEngineResult.SearchedLocations);
+            }
```

Comment:
Doing inside EnsureSuccessful will lose context on what the view engine result is for (like if the view is for partial or not) and also in the latest iteration we have cut down logging in lot of places except top level concepts like ViewResult and PartialResult.